src/foam/dao/HTTPSink.java

```diff
@@ -15,28 +15,32 @@
 import foam.nanos.http.Format;
 import foam.util.SafetyUtil;
 
+import javax.crypto.Mac;
 import javax.servlet.http.HttpServletResponse;
 import java.io.*;
 import java.net.HttpURLConnection;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
+import org.apache.commons.codec.binary.Base64;
 
 public class HTTPSink
     extends AbstractSink
 {
   protected String url_;
   protected String bearerToken_;
+  protected String payloadSignature_;
   protected Format format_;
   protected PropertyPredicate propertyPredicate_;
   protected boolean outputDefaultValues_;
 
   public HTTPSink(String url, Format format) {
-    this(url, "", format, null, false);
+    this(url, "", "", format, null, false);
   }
 
-  public HTTPSink(String url, String bearerToken, Format format, PropertyPredicate propertyPredicate, boolean outputDefaultValues) {
+  public HTTPSink(String url, String bearerToken, String payloadSignature, Format format, PropertyPredicate propertyPredicate, boolean outputDefaultValues) {
```

Comment:
we don't want to pass in the payload signature .. we want to compute it from the outputter.stringfy(obj) string .. Intuit will be providing us with a shared secret that we should be using to generate the hash.